src/ResourceManagement/Sql/SqlEncryptionProtectorImpl.cs

```diff
@@ -0,0 +1,221 @@
+            this.resourceGroupName = resourceGroupName;
+            this.sqlServerName = sqlServerName;
+            this.serverKeyName = innerObject?.Name;
+        }
+
+        /// <summary>
+        /// Creates an instance of external child resource in-memory.
+        /// </summary>
+        /// <param name="innerObject">Reference to the inner object representing this external child resource.</param>
+        /// <param name="sqlServerManager">Reference to the SQL server manager that accesses firewall rule operations.</param>
+        ///GENMHASH:4DD6445361F09E0B185DE0E8D24170D3:81D0EDB620000F6B86DC1403C1548E6A
+        internal SqlEncryptionProtectorImpl(EncryptionProtectorInner innerObject, ISqlManager sqlServerManager)
+            : base(innerObject, null)
+        {
+            this.sqlServerManager = sqlServerManager;
+            if (innerObject != null && innerObject.Id != null)
+            {
+                if (innerObject.Id != null)
+                {
+                    ResourceId resourceId = ResourceId.FromString(innerObject.Id);
+                    this.resourceGroupName = resourceId.ResourceGroupName;
+                    this.sqlServerName = resourceId.Parent.Name;
+                }
+            }
+            this.serverKeyName = innerObject?.Name;
+        }
+
+        public override string Name()
+        {
+            return this.Inner?.Name;
+        }
+
+        ///GENMHASH:E9EDBD2E8DC2C547D1386A58778AA6B9:7EBD4102FEBFB0AD7091EA1ACBD84F8B
+        public string ResourceGroupName()
+        {
+            return this.resourceGroupName;
+        }
+
+        ///GENMHASH:6485A67119B54B835368ADA812A96C10:EEFF62BBC201203BB09A9D9FE63B14EC
+        public string ServerKeyName()
+        {
+            return this.Inner.ServerKeyName;
+        }
+
+        ///GENMHASH:61F5809AB3B985C61AC40B98B1FBC47E:03B7E74B9CEC907229FC7E3E3FC8EFC4
+        public string SqlServerName()
+        {
+            return this.serverKeyName;
+        }
+
+        ///GENMHASH:C4C0D4751CA4E1904C31CE6DF0B02AC3:B30E59DD4D927FB508DCE8588A7B6C5E
+        public string Kind()
+        {
+            return this.Inner.Kind;
+        }
+
+        ///GENMHASH:8F04665E49050E6C5BD8AE7B8E51D285:48C470FF98C85D0D802E5FBBD537EFBE
+        public string Thumbprint()
+        {
+            return this.Inner.Thumbprint;
+        }
+
+        ///GENMHASH:6BCE517E09457FF033728269C8936E64:40A980295F5EA8FF8304DA8C06E899BF
+        public SqlEncryptionProtectorImpl Update()
+        {
+            return this;
+        }
+
+        ///GENMHASH:39E79EEEBE800C0263D7785D2CDD0C8F:AA25005E7A9BAF367EA281FE0DB49192
+        public string Uri()
+        {
+            return this.Inner.Uri;
+        }
+
+        ///GENMHASH:5AD91481A0966B059A478CD4E9DD9466:2EA6F22A1E9075415416C51D51B2ED7C
+        protected async Task<Models.EncryptionProtectorInner> GetInnerAsync(CancellationToken cancellationToken = default(CancellationToken))
```

Comment:
shouldn't this be an overriden method?

Reply:
not in this case; the Impl inherits from ChildResource<> which does not define it.